src/main/java/de/tum/in/www1/artemis/domain/quiz/DragAndDropQuestion.java

```diff
@@ -133,41 +132,23 @@ public Boolean isValid() {
             return false;
         }
 
+        // A drag item can either be a text or a picture, but not both or none
+        for (DragItem dragItem : dragItems) {
+            if (StringUtils.isEmpty(dragItem.getText()) == StringUtils.isEmpty(dragItem.getPictureFilePath())) {
+                return false;
+            }
+        }
+
         // check if at least one correct mapping exists
         return getCorrectMappings() != null && !getCorrectMappings().isEmpty();
 
-        // TODO (?): Add checks for "is solvable" and "no misleading correct mapping" --> look at the implementation in the client
+        // TODO: (?) Add checks for "is solvable" and "no misleading correct mapping" --> look at the implementation in the client
     }
 
-    /*
-     * NOTE: The file management is necessary to differentiate between temporary and used files and to delete used files when the corresponding question is deleted or it is
-     * replaced by another file. The workflow is as follows 1. user uploads a file -> this is a temporary file, because at this point the corresponding question might not exist
-     * yet. 2. user saves the question -> now we move the temporary file which is addressed in backgroundFilePath to a permanent location and update the value in backgroundFilePath
-     * accordingly. => This happens in @PrePersist and @PostPersist 3. user might upload another file to replace the existing file -> this new file is a temporary file at first 4.
-     * user saves changes (with the new backgroundFilePath pointing to the new temporary file) -> now we delete the old file in the permanent location and move the new file to a
-     * permanent location and update the value in backgroundFilePath accordingly. => This happens in @PreUpdate and uses @PostLoad to know the old path 5. When question is deleted,
-     * the file in the permanent location is deleted => This happens in @PostRemove
-     */
-
     /**
-     * Initialisation of the DragAndDropQuestion on Server start
+     * This method is called after the entity is saved for the first time. Before creation, we don't know yet the id of the entity, so we use a placeholder in the
+     * backgroundFilePath
```

Comment:
```suggestion
     * This method is called after the entity is saved for the first time. Before creation, we didn't knew the id of the entity, so we used a placeholder in the
     * backgroundFilePath that is now replaced
```

Reply:
That does not make sense to me. I'm not narrating something that happened (past tense) but describing something in a created scenario (present tense).